kythe/typescript/README.md

```diff
@@ -1,5 +1,241 @@
 # Kythe indexer for TypeScript
 
+## VName Specification
+
+This specification defines how the indexer expresses VNames for TypeScript
+declarations. You may find this useful if you are developing an application that
+relies on TypeScript code you don't want to re-index.
+
+[spec tests](./testdata/declaration_spec.ts)
+
+### VName signature
+
+The signature description language used in this document is similar to regex.
+
+-   A substring in `\b\$.*\b` is a variable.
+    -   e.g. `$DECLARATION_NAME` is a variable refering to the name of
+        declaration.
+-   A substring in `\[.*\]\?` is matched 0 or 1 times.
+    -   e.g. `(hidden)?` is really `hidden` or ``.
+-   A substring in `\[.*\]\*` is matched 0 or more times.
+    -   e.g. `[a]*` is ``, or`a`, or`aa`, ...
+-   All other substrings are literals.
+    -   e.g. `get#$NAME` is really `get#foo` if `$NAME = foo`.
+
+The signature of a TypeScript declaration is defined by the following schema:
+
+```regex
+$PART[.$PART]*[#type]?
+```
+
+where `$PART` is a component of the enclosing declaration scope and `#type` is
+appended to the signature of type declarations. `SyntaxKind`s that are type
+declarations are:
+
+-   `ClassDeclaration` (also a value)
+-   `EnumDeclaration` (also a value)
+-   `InterfaceDeclaration`
+-   `TypeAliasDeclaration`
+-   `TypeParameter`
+
+As an example of this schema, in
+
+```typescript
+class A {
+  public foo: string;
+}
+
+type B = A;
+```
+
+`foo` has the signature `A.foo` and `B` has the signature `B#type`.
```

Comment:
Should it be some A#instance.foo, given that A can have static members too?
Maybe it's fine to just document the current behavior.

Reply:
Yeah, this is actually something we need to resolve. Let's point static members to `A#type.foo` and instance members to `A.foo`, as you suggested.

Reply:
Resolved in 74e1e4a2771a7c58ddbd56a00fcf0505b6c082a1

Reply:
This is now A#type.foo, right?